client/src/main/java/io/pravega/client/connection/impl/AppendBatchSizeTrackerImpl.java

```diff
@@ -50,8 +65,7 @@ public void recordAppend(long eventNumber, int size) {
         long now = Math.max(lastAppendTime.get(), clock.get());
         long last = lastAppendTime.getAndSet(now);
         lastAppendNumber.set(eventNumber);
-        millisBetweenAppends.addNewSample(now - last);
-        appendsOutstanding.addNewSample(eventNumber - lastAckNumber.get());
+        nanosBetweenAppends.addNewSample(now - last);
```

Comment:
Why are we dropping `appendsOutstanding.addNewSample(eventNumber - lastAckNumber.get());` from this method? Why this was before?

Reply:
It performs better (in terms of cpu use, this showed up on a flamegraph) and provides a more consistent value if 'appendsOutstanding' is defined in a uniform way as opposed to being an mixture of two different ways of measuring. 

Before this method was called from the netty thread after the handoff had occurred. Now there is no netty thread so this is being called in-line in the write path. So the meaning of the 'outstanding' is subtly different. So I decided to simplify things and remove it from the hot path, by only sampling the value after an ack. 

This has the effect that rather than attempting to average a sawtooth distribution we are just sampling the low points. Hence this will have the effect of lowering the equivalent batch size to below where it was before. However because we are including more data in the definition of being 'outstanding' by virtue of having eliminated the netty thread, this may end up being a wash.